src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -17,24 +17,10 @@ public abstract partial class JsonConverter<T> : JsonConverter
         /// <summary>
         /// When overridden, constructs a new <see cref="JsonConverter{T}"/> instance.
         /// </summary>
-        protected internal JsonConverter() : this(initialize: true)
-        { }
-
-        internal JsonConverter(bool initialize)
+        protected internal JsonConverter()
```

Comment:
I think HandleNull logic might need some more love as well. Currently we're modifing `HandleNullOnRead` when `HandleNull` is read (unless it's overriden) and then we reassign it here again. Also there is a high chance HandleNull will be actually correctly set in the ctor which means after this base ctor is called.

Reply:
Yeah, I tried to improve that as well but couldn't come up with a good way that preserves the existing invariants. Decided to leave as-is for now.